src/H5T.c

```diff
@@ -789,6 +854,22 @@ H5T__init_inf(void)
     }     /* end if */
 #endif
 
+#ifdef H5_HAVE_COMPLEX_NUMBERS
+    /*
+     * A complex number is considered to be infinite if any of its parts
+     * (real or imaginary) is infinite, so simply set the real part of
+     * the complex number positive/negative infinity values to the
+     * positive/negative infinity value for the base floating-point type.
+     * The imaginary part should be zeroed-out bits.
+     */
+    memcpy(&H5T_NATIVE_FLOAT_COMPLEX_POS_INF_g, &H5T_NATIVE_FLOAT_POS_INF_g, sizeof(float));
+    memcpy(&H5T_NATIVE_FLOAT_COMPLEX_NEG_INF_g, &H5T_NATIVE_FLOAT_NEG_INF_g, sizeof(float));
+    memcpy(&H5T_NATIVE_DOUBLE_COMPLEX_POS_INF_g, &H5T_NATIVE_DOUBLE_POS_INF_g, sizeof(double));
+    memcpy(&H5T_NATIVE_DOUBLE_COMPLEX_NEG_INF_g, &H5T_NATIVE_DOUBLE_NEG_INF_g, sizeof(double));
+    memcpy(&H5T_NATIVE_LDOUBLE_COMPLEX_POS_INF_g, &H5T_NATIVE_LDOUBLE_POS_INF_g, sizeof(long double));
+    memcpy(&H5T_NATIVE_LDOUBLE_COMPLEX_NEG_INF_g, &H5T_NATIVE_LDOUBLE_NEG_INF_g, sizeof(long double));
+#endif
```

Comment:
This seems odd.   Why are the complex infinities not of _Complex type?   If it's correct, then why not just use the regular float, double, etc, types and not define complex infinities?

Reply:
They are of `_Complex` type, but only their real part gets set to +/- Infinity. See the comment above these lines.

Reply:
OK, I see that they are _Complex, and I understand the standard's behavior.  It's the way that only half of the value is being initialized that was concerning me.  I suppose that it's in the C89 standard to have the other half zero-initialized, so should be fine.